base/array.hpp

```diff
@@ -81,8 +82,8 @@ struct BoundedArray final {
   bool empty() const;
   std::size_t size() const;
 
-  std::array<Element, size_> data;
-  std::int32_t actual_size;
+  std::array<Element, max_size> data;
+  std::int32_t size_;
```

Comment:
This is weird.  We don't normally use `_` for members of a `struct`.  I suspect that this could a `class` (with private members) now.